### ISSUE TITLE
Issue with missing limbs

### ISSUE THREAD

**Issue by Vibecoder9000** [hyz317/StdGEN#7]
I've noticed an issue with missing limbs. Most commonly, feet. Much of the images generally available are 3/4 type shots. Without the feet in the image, it results in a unusable posed result. I can think of a way to fix this: add feet via fast image generation, pixart is fast and light. Otherwise it's a issue with the model that processes the posing, and I don't have any idea how that really works. Accuracy for that step alone is not great and there's probably some easy improvement there.

**Comment by hyz317** [comment]
Could you please provide an example input image where the problem occurs? This will help us investigate further. In our testing, the model has generally performed well for most in-the-wild cases.

**Comment by Vibecoder9000** [comment]
Without:

![Image](https://github.com/user-attachments/assets/85011dbe-cbc7-4d2e-9df2-e3c13a51479d)

![Image](https://github.com/user-attachments/assets/a6ded248-97aa-4821-8eb3-f55b4c226a69)

![Image](https://github.com/user-attachments/assets/1608706e-d02c-4cd8-ba8a-36f4454a6c29)

![Image](https://github.com/user-attachments/assets/36efb635-54d4-47d6-8b6b-99020a0055c7)

With feet+legs:

![Image](https://github.com/user-attachments/assets/4ba4ce07-35a8-49c3-befd-d18587f6a098)

![Image](https://github.com/user-attachments/assets/05627c3d-381b-45e4-825c-af6da65e7528)

It's a bit of a mixed bag. There's some inaccuracy caused by the missing legs, but not enough to explain the other sample "without" images' results. Is there a better way to use this?

**Comment by hyz317** [comment]
Thank you for your detailed feedback and for sharing your observations!

Currently, the automatic segmentation in our demo relies on rm_anime_bg (https://github.com/shirayu/rm_anime_bg), which works well for anime-style images but may not perform as effectively for other styles like 2.5D or real-world images. If the background isn’t fully removed, it can significantly impact the results. For better results, you could try using other background removal tools like Clipdrop before uploading the image. Our method does work for real human and other styles, as demonstrated in the paper.

Additionally, our training data consists of full-body images, so inputting half-body photos (e.g., missing feet or legs) may lead to suboptimal results. You could experiment with tools like Clipdrop’s image uncrop feature to extend the image to a full-body format. We’ll also consider supporting more diverse inputs, such as half-body photos, as part of our future work.

**Comment by Vibecoder9000** [comment]
I find using Google's new multimodal flash model works well for background removal. It produces perfect results with a bit of prompting. (ask it what objects there are, tell it your goal, ask it to meet that) Issues: hallucinating which happens rarely and image degradation because of resolution and vae channels altering small details. It can also zoom out images to better match the remaining images but there's a significant hallucination percentage.

**Comment by hyz317** [comment]
Thanks for sharing this helpful tip! I'll add it to the README to help other users.
